scripts/core/superdesk-api.d.ts

```diff
@@ -75,6 +75,7 @@ declare module 'superdesk-api' {
                     onSpikeMultiple?(items: Array<IArticle>): Promise<onSpikeMiddlewareResult>;
                     onPublish?(item: IArticle): Promise<onPublishMiddlewareResult>;
                     onRewriteAfter?(item: IArticle): Promise<IArticle>;
+                    onSend?(desk: IDesk, itemIds: Array<string>): Promise<void>;
```

Comment:
I'd call it `onSendBefore` so it's clear when the code is getting executed.

I'd also have `itemIds` argument come first in the function signature so it reads more naturally like "send those items to this desk".

What are `itemIds` by the way? Shouldn't it accept a single `item: IArticle` instead?

Reply:
This action can be used with multi-select as well, in which case the directive provides the ids of the items.